src/reflection/FStar.Reflection.NBEEmbeddings.fs

```diff
@@ -628,13 +685,16 @@ let e_sigelt_view =
             BU.bind_opt (unembed (e_list e_ctor) cb dcs) (fun dcs ->
             Some <| Sg_Inductive (nm, us, bs, t, dcs))))))
 
-        | Construct (fv, _, [(t, _); (ty, _); (univs, _); (fvar, _); (r, _)]) when S.fv_eq_lid fv ref_Sg_Let.lid ->
+        | Construct (fv, _, [(lbs, _); (r, _)]) when S.fv_eq_lid fv ref_Sg_Let.lid ->
             BU.bind_opt (unembed e_bool cb r) (fun r ->
-            BU.bind_opt (unembed e_fv cb fvar) (fun fvar ->
-            BU.bind_opt (unembed e_univ_names cb univs) (fun univs ->
-            BU.bind_opt (unembed e_term cb ty) (fun ty ->
+            BU.bind_opt (unembed (e_list e_letbinding) cb lbs) (fun lbs ->
+            Some <| Sg_Let (r, lbs)))
+
+        | Construct (fv, _, [(t, _); (us, _); (nm, _)]) when S.fv_eq_lid fv ref_Sg_Val.lid ->
```

Comment:
Is this code fixing the unembedding of `Sg_val`? Was it a bug in the previous code? (Asking only because doesn't seem directly related to the change that this PR is doing.)

Reply:
Yes, it's fixing the `Sg_Val` case. I really can’t remember now, but I'm almost positive it was a bug.